### DISCUSSION TITLE
Question on "respond with HTML, not JSON" / partial or complete HTML?

### DISCUSSION THREAD

**Discussion by vguhesan** [bigskysoftware/htmx#1557] [Q&A] [answered]
When the documentation [here](https://htmx.org/docs/#introduction) states:
"Note that when you are using htmx, on the server side you typically **respond with HTML, not JSON.** This keeps you firmly within the original web programming model, using Hypertext As The Engine Of Application State without even needing to really understand that concept." 

Do you typically respond the block of partial content you want as a complete HTML (meaning with `<html><body>{{partial-content}}</body></html>` or just the `{{partial-content}}`? 

What is the standard? It would be nice if there was some clarification on this. As a back-end developer. I feel that if I had the full document including the `<html><body>...</body><html>` will make it ideal to edit and modify (as well as render/preview inside IDEs) it as a partial with a full view as you're editing.

Trying to understand where the the community stands on this as a standard or as best practices. 

Thank you.

**Comment by Telroshan** [comment, marked as answer]
To each their own I guess, but I personnally only send `{{partial-content}}` when applicable
- There's no point in sending a whole page's HTML if you were only going to use a small portion of it
- The backend would spend unnecessary time & resources fetching any required data & generating all the parts of the generated HTML response, that wouldn't even be used by the client as only the `{{partial-content}}` would be _(the rest would be ditched away)_
- By using a templating system in your backend, you won't have to duplicate any code but simply factor your `{{partial-content}}` in its own template/component/whatever, and use that in your bigger page as well as your partial content routes
- I understand the concern about visibility when editing, though I would compare those to functions in your backend. It may be a little bit contrived, but bear with me on this analogy:
When you are going to reuse a piece of code in your backend _(say a for loop with some specific logic inside)_, you don't call the whole surrounding function that initially contained that piece of code, to have the "full view" or "full context" where it is to be reused, when you only needed to copy that small portion _(the for loop)_.
Assuming here that you would make a separate function to factor that logic, it wouldn't be a problem to edit, since you can quickly head to the definition of that new-factored function, as well as find its usages.
I think the same principle applies to factoring parts of your HTML into separated "components" or "templates". It's like factoring a piece of code you want to reuse, as its own separated thing, then you can quickly find the possible surrounding contexts & usages, using a simple Ctrl + shift + F or whatever similar _(maybe your IDE even supports a Ctrl + click there to jump to the referenced templates directly)_.

**Reply by vguhesan** [reply]
Thank you. Your points are all valid. Much appreciated.
